### PR TITLE
acceptance: allow docker tests to run

### PR DESCRIPTION
**Pull request by madelynnblue**
Cache the builder image, which is now just one image with all langs.

Continuation of #4113

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4301)

### CODE REVIEW

**Comment by bdarnell** [comment]
While we're adding more caching can we cache the `docker-spy` image too?

File: build/circle-deps.sh [hunk 1]
```diff
@@ -9,6 +9,26 @@ function is_shard() {
...
 }
 
+function fetch_docker() {
+  local image=${1}
+  local tag=${2}
+  local name=${image}:${tag}
+  if ! docker images | grep -q "${name}"; then
+    # If there's a base image cached, load it. A click on CircleCI's "Clear
+    # Cache" will make sure we start with a clean slate.
+    builder="${builder_dir}/builder.${name}.tar"
```
Comment (bdarnell) [inline build/circle-deps.sh]: `builder` is no longer the right name for this variable. I would say `image` if you weren't using that name above.

**Comment by mjibson** [comment]
I added a TODO to cache the docker-spy image.

---

Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[build/circle-deps.sh, line 19 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAC4M7EndAgemWWapXf-r1-19:-KACEMrPWNIoKbWJZ2Fx:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7f95d70b800df51eaac7a325e00f8534cf8e45d/build/circle-deps.sh#L19)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-:-KACE1zAis2B7NYInEs9:-923966415)*

**Comment by bdarnell** [comment]
LGTM

**Comment by tamird** [comment]
Reviewed 6 of 7 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KACGw7HQTkp5Cs1teyV:261807131) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
why remove the tag argument? if we expect to never have to pull these images because they're always cached, we need to remove that code. This basically leaves things in a state where running these locally can fetch the wrong version of this image.

---

*[acceptance/ruby_test.go, line 27 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGhSdTIoI7iK6eG09:-KACGhSdTIoI7iK6eG0A:-1042013254) ([raw file](https://github.com/cockroachdb/cockroach/blob/c8063786a6e35bd493160e8f294547112cc0bd90/acceptance/ruby_test.go#L27)):*
can you make this of the form "TODO(mjibson): ...."?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by mjibson** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[acceptance/ruby_test.go, line 27 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGhSdTIoI7iK6eG09:-KACM6-Kmh_GjDK0AXzF:1637509395) ([raw file](https://github.com/cockroachdb/cockroach/blob/c8063786a6e35bd493160e8f294547112cc0bd90/acceptance/ruby_test.go#L27)):*
Oops, this shouldn't be there. Will remove.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by mjibson** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KACMNbBBJC41SkeCigi:988396316) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
The tag argument was to specify the language to use. They've all been combined into a single image, so that can go away.

Now it implies using `:latest`, and the build script fetches a specific , dated tag, and then calls that latest. Since running locally that script isn't being used, it'll pull whatever the `latest` tag is from the docker hub. So when we build and push new versions of the image to the hub, we'll also have to tag and push as `latest`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by tamird** [comment]
Review status: 6 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KACMvjfVEONhx8hwy5E:-1915783390) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
sorry, this comment was anchored to the wrong line. I really just want to retain the `tag` argument to `testDocker`, with a hardcoded constant here that is equal to what's in circle-deps.sh.

Basically, the current state prevents reproducible builds. If I check out an old ref, I'll ge the wrong image because `latest` has drifted.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by mjibson** [comment]
Review status: 5 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KACOJZA5hsjreimrqik:285382910) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
Agreed. I've hard coded the tag into the image name.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by tamird** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KACPV27W-Iwu5e46nja:836193736) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
You need to keep the separate argument. Note the change you end up making to the `ImagePull` call.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KACPlgwjjqsjY1PPe6f:-1871144763) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
Err, perhaps not an argument, but you at least need a separate binding.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by mjibson** [comment]
Refactored the docker event handling to allow one shot containers to die.

---

Review status: 6 of 8 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-:-KAH9vR-K6TydVe0WLJK:-276761581)*

**Comment by mjibson** [comment]
Review status: 6 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KAHA1y6L5toDpKfACUe:1095622836) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
I guess I still don't understand. Does your point still stand based on the current code?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by tamird** [comment]
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[acceptance/cluster/localcluster.go, line 125 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHAoZh_S-BVQ6PFJwU:-KAHAoZh_S-BVQ6PFJwV:356643234) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L125)):*
why not make this a `oneshot *Container`

---

*[acceptance/cluster/localcluster.go, line 179 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHAB2BqSSt7VaeV1ZT:-KAHAB2BqSSt7VaeV1ZU:-1720926110) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L179)):*
Why can't you use `pullImage` here? You're duplicating a lot of code.

---

*[acceptance/cluster/localcluster.go, line 198 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHAO91hv13MD2In9CZ:-KAHAO91hv13MD2In9C_:122900722) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L198)):*
ditto, why can't you use `createContainer`?

---

*[acceptance/cluster/localcluster.go, line 210 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHASQQdADDd9UqU20t:-KAHASQQdADDd9UqU20u:1975599415) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L210)):*
If you use `createContainer` above, you can use `(*Container).Remove()` here.

---

*[acceptance/cluster/localcluster.go, line 217 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHA_Dqi0qMxQwlSceM:-KAHA_Dqi0qMxQwlSceN:1993168358) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L217)):*
`(*Container).Start()`

---

*[acceptance/cluster/localcluster.go, line 220 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHAi842eMXrwdllfgp:-KAHAi842eMXrwdllfgq:-1632877887) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L220)):*
`(*Container).Wait()`

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KAHAsmeoMy8BGkA_4uw:-1617661061) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
Yes, you need to call ImagePull with a Tag in the options. I can show you in person, if you don't know what I'm talking about.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by mjibson** [comment]
Review status: 6 of 8 files reviewed at latest revision, 8 unresolved discussions.

---

*[acceptance/cluster/localcluster.go, line 125 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHAoZh_S-BVQ6PFJwU:-KAJPrEXelbJaWIZ0wAA:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L125)):*
Done.

---

*[acceptance/cluster/localcluster.go, line 179 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHAB2BqSSt7VaeV1ZT:-KAJPrc1evsD4TY_zzCB:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L179)):*
Done.

---

*[acceptance/cluster/localcluster.go, line 198 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHAO91hv13MD2In9CZ:-KAJaM3oPBQ0iv4WNLPS:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L198)):*
Done.

---

*[acceptance/cluster/localcluster.go, line 210 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHASQQdADDd9UqU20t:-KAJaNwkSjqT_gxfReIn:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L210)):*
Done.

---

*[acceptance/cluster/localcluster.go, line 217 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHA_Dqi0qMxQwlSceM:-KAJaOT0B2VRHPZRJDq1:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L217)):*
Done.

---

*[acceptance/cluster/localcluster.go, line 220 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAHAi842eMXrwdllfgp:-KAJaOp9qNFuiKvj7ybR:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/51b96da301fda39e63924bae050648daa426f200/acceptance/cluster/localcluster.go#L220)):*
Done.

---

*[acceptance/main_test.go, line 201 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KACGw7HQTkp5Cs1teyU:-KAJgNiYa9Xwl69mvBHT:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/10339741e6c9eb86abb6a4ce9de8609dbd574ae9/acceptance/main_test.go#L201)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by tamird** [comment]
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[acceptance/cluster/localcluster.go, line 177 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKHW_yPYrcOUbj0c_K:-KAKHW_zjQ0lWGSWr7O8:-1249652531) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L177)):*
make the first argument of type `types.ImagePullOptions`. Also, can you follow th naming used elsewhere? s/c/containerConfig/, s/h/hostConfig/?

---

*[acceptance/cluster/localcluster.go, line 186 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKHoOEqXWpjhg3T3KB:-KAKHoOFgYLh44VnzW_m:1139481469) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L186)):*
you should add a "name string" argument to this function so you can pass it into createContainer.

---

*[acceptance/cluster/localcluster.go, line 450 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKI8heD10HlgWMXtUc:-KAKI8heD10HlgWMXtUd:-164202530) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L450)):*
this logic needs a comment

---

*[acceptance/main_test.go, line 205 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKIEkoiQ0WtdtvtuYG:-KAKIEkoiQ0WtdtvtuYH:1737873603) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/main_test.go#L205)):*
Errorf

---

*[acceptance/main_test.go, line 214 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKIGPfNLjT0L_GXvhW:-KAKIGPfNLjT0L_GXvhX:298328377) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/main_test.go#L214)):*
s/c/containerConfig/

---

*[acceptance/main_test.go, line 223 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKIIBUtBRHG8YYDAo4:-KAKIIBUtBRHG8YYDAo5:882752253) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/main_test.go#L223)):*
s/h/hostConfig/

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[acceptance/cluster/localcluster.go, line 182 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KALIkdhvbkjkXUmDDdr:-KALIkdhvbkjkXUmDDds:-65433816) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L182)):*
what happens if you pull and the image is already present?

---

*[acceptance/cluster/localcluster.go, line 185 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KALIdJ4WxzB2hRjEB3X:-KALIdJ4WxzB2hRjEB3Y:-1019808947) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L185)):*
can you do this in the caller?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by mjibson** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[acceptance/cluster/localcluster.go, line 177 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKHW_yPYrcOUbj0c_K:-KAfVrxCWUDLtLbPixx4:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L177)):*
Done.

---

*[acceptance/cluster/localcluster.go, line 182 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KALIkdhvbkjkXUmDDdr:-KAfVsEi9JrZ4jF5Tl3R:-256207208) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L182)):*
It immediately returns successfully.

---

*[acceptance/cluster/localcluster.go, line 185 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KALIdJ4WxzB2hRjEB3X:-KAfW3A1wxo3yongpBTS:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L185)):*
Done.

---

*[acceptance/cluster/localcluster.go, line 186 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKHoOEqXWpjhg3T3KB:-KAfW7UHFgcwjGiUwOO0:-1409813898) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L186)):*
Why does it need a name? If unspecified it gets a random one.

---

*[acceptance/cluster/localcluster.go, line 450 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKI8heD10HlgWMXtUc:-KAfWPPq42g-fUVDjRK_:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L450)):*
Done.

---

*[acceptance/main_test.go, line 205 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKIEkoiQ0WtdtvtuYG:-KAfWVOVJ22rscJ-3QSb:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/main_test.go#L205)):*
Done.

---

*[acceptance/main_test.go, line 214 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKIGPfNLjT0L_GXvhW:-KAfW_9ZdMgG3fr4HOVH:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/main_test.go#L214)):*
Done.

---

*[acceptance/main_test.go, line 223 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKIIBUtBRHG8YYDAo4:-KAfWZx9LUhBPGcAabqa:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/main_test.go#L223)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by tamird** [comment]
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[acceptance/cluster/localcluster.go, line 182 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KALIkdhvbkjkXUmDDdr:-KAfbM0o6_4SBI77C_2S:-1400672260) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L182)):*
Ah, interesting. We should apply that knowledge to other callers of `pullImage` which currently have awkward retry logic.

---

*[acceptance/cluster/localcluster.go, line 186 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKHoOEqXWpjhg3T3KB:-KAfbUWJXIp84jp9UE-j:-1914635850) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L186)):*
Right, but why not specify it? You're just making it harder to inspect what's going on with standard tools e.g. `docker ps`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by mjibson** [comment]
Review status: 1 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/cluster/localcluster.go, line 182 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KALIkdhvbkjkXUmDDdr:-KAfeWcQSi48aX4HrRv_:1404480145) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L182)):*
Created #4410 to track this.

---

*[acceptance/cluster/localcluster.go, line 186 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301#-KAKHoOEqXWpjhg3T3KB:-KAffhmBUYvWUsSdAAI-:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9f14c812bf06bcd4dd09d790fd5bdfa029cad89/acceptance/cluster/localcluster.go#L186)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*

**Comment by tamird** [comment]
Reviewed 7 of 7 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4301)*
